lib/ansible/modules/cloud/amazon/ec2_group_facts.py

```diff
@@ -147,17 +147,13 @@ def main():
     except ClientError as e:
         module.fail_json(msg=e.message, exception=traceback.format_exc())
 
-    # Turn the boto3 result in to ansible_friendly_snaked_names
-    snaked_security_groups = []
+    # Modify boto3 tags list to be ansible friendly dict and then camel_case
+    camel_security_groups = []
     for security_group in security_groups['SecurityGroups']:
-        snaked_security_groups.append(camel_dict_to_snake_dict(security_group))
+        security_group['Tags'] = boto3_tag_list_to_ansible_dict(security_group['Tags'])
+        camel_security_groups.append(camel_dict_to_snake_dict(security_group))
```

Comment:
Should we have a helper that does `camel_list_to_snake_list` so that I don't have to suggest turning this into a list comprehension every time?